cumulus/ansible/tasks/playbooks/roles/ec2-pod/tasks/launch.yml

```diff
@@ -0,0 +1,149 @@
+---
+# Note that we don't include master/node ami variables - these
+# could be conditionally checked, but because the next thing we
+# do is try to find the master/node ami if they aren't defined
+# it seems unnecessary.
+- name: Fail early if required variables are not defined
+  fail:
+    msg: "Variable {{ item }} is not defined"
+  when: item not in hostvars[inventory_hostname]
+  with_items:
+    - aws_keyname
+    - master_instance_type
+    - node_instance_type
+    - node_instance_count
+    - cluster_id
+    - cluster_region
+    - cluster_zone
+
+
+- name: Define default firewall rules if not provided
+  set_fact:
+    ec2_pod_rules:
+      - proto: tcp # ssh
+        from_port: 22
+        to_port: 22
+        cidr_ip: "{{ ssh_cidr_ip }}"
+
+      - proto: all
+        group_name: "ec2_pod_{{cluster_id}}"
+        group_desc: "Ec2 security group for cluster {{ cluster_id }}"
+  when: ec2_pod_rules is not defined
+
```

Comment:
Yes,  currently this removes the extra_rules  variable - instead you should pass in your own custom ```ec2_pod_rules``` list that looks like:

```yaml
ec2_pod_rules:
  - proto: tcp # ssh
    from_port: 22
    to_port: 22
    cidr_ip: "{{ ssh_cidr_ip }}"

  - proto: "tcp"
    from_port: 7000
    to_port: 7000
    cidr_ip: "source_ip"

  - proto: all
    group_name: "ec2_pod_{{cluster_id}}"
    group_desc: "Ec2 security group for cluster {{ cluster_id }}"
```

Unless this is overly burdensome in which case I can add ```extra_vars``` back in - my feeling though is that even though this is a little more typing,  it is more explicit,  and prevents edge cases produced by merging rule sets.